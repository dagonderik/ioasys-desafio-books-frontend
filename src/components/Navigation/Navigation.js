import React from 'react';

const Navigation = ({ onRouteChange, user }) =>{
    return(
    <nav style={{display: 'flex', justifyContent: 'space-between'}}>
        <p className="black  f1  "><b>ioasys</b> Books</p>
        <p onClick={() => onRouteChange('login')} className='f3  dim black pa3 pointer'>Bem vindo, {user.data.name}! <span role='img' aria-label='Logout'>ðŸ”™</span></p>
    </nav>
    )
}


export default Navigation;