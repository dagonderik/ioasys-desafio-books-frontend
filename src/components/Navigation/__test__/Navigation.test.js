import React from "react";
import ReactDOM from 'react-dom';
import Navigation from "../Navigation";
import renderer from "react-test-renderer";

const userTest = {
    data:{
        name:"name"
    }
}

it("renders without crashing", () => {  
    const div = document.createElement("div");
    ReactDOM.render(<Navigation
    onRouteChange={()=>{}}
    user={userTest}></Navigation>,div)
})

it("matches snapshot", () =>{
    const tree = renderer.create(<Navigation
        onRouteChange={()=>{}}
        user={userTest}></Navigation>).toJSON();
    expect(tree).toMatchSnapshot();
})