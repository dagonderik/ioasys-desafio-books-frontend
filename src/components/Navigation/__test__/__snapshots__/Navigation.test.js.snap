// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
    }
  }
>
  <p
    className="black  f1  "
  >
    <img
      alt="logo"
      className="pr2"
      src="logo2.png"
      style={
        Object {
          "height": 50,
          "left": 10,
          "position": "relative",
          "top": 11,
          "width": 150,
        }
      }
    />
    Books
  </p>
  <p
    className="f3 black pa3"
  >
    Bem vindo, 
    name
    !
    <span
      aria-label="Logout"
      className="dim pointer"
      onClick={[Function]}
      role="img"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M5 11h8v2H5v3l-5-4 5-4v3zm-1 7h2.708a8 8 0 1 0 0-12H4A9.985 9.985 0 0 1 12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10a9.985 9.985 0 0 1-8-4z"
          />
        </g>
      </svg>
    </span>
  </p>
</nav>
`;
